### ISSUE TITLE
[Accessibility][CommandBar]: Disabled Menu items should be focusable

### ISSUE THREAD

**Issue by satishc2437** [OfficeDev/office-ui-fabric-react#912]
Hi,
We are using the CommandBar control in our view for buttons such as "Save", "Queue", etc. These are typically important buttons and will show up at the top of the page. The buttons support enable/disable behavior based on the error state of the page.

We found that a disabled button in the command bar doesn't gets focus while using the Arrow keys. It directly jumps to the next enabled item.

This behavior is violating the ARIA design pattern for Menu bar. Following is the quote from WAI-ARIA-Practices-1.1 page (http://www.w3.org/TR/wai-aria-practices-1.1/#menu)

1. Disabled menu items are focusable but cannot be activated.

![image](https://cloud.githubusercontent.com/assets/13657916/22460780/7f9db928-e7cc-11e6-8267-4b26698b2200.png)

Requesting you fix this issue as appropriate.

Thanks,
Satish

**Comment by dzearing** [comment]
Thanks for pointing this out; it's confusing because every application, including browser behavior, defaults to making disabled buttons non-tabstops. All the applications I've used skip over disabled buttons. And if you disable a button in the browser and set tabindex=0, it will still skip over it.

This implies that:

1. Most apps are implementing it wrong
2. We have to fight the browser to do the right thing
3. Even if we do it right, every other app will have done it wrong

I'm not opposed to fixing so don't read this wrong. But there is a much larger discussion to be had here. I've looped in the Edge accessibility contact we have so see what their take is.

**Comment by mdahamiwal** [comment]
@dzearing, i think having something like https://jqueryui.com/menu/ will help. We don't need tabindex=0 as menu bar defines navigation using arrow keys.

**Comment by dzearing** [comment]
@mdahamiwal not sure i follow. You need focus to be set on the active item for screenreader default behavior but also just in general for a variety of other reasons. But for disabled buttons, we just can't combine the native disabled state with focusability.

We already fake disabled states for anchors (Fabric Button with href, for example) because browsers don't natively support disabled anchor tags. So it wouldn't be difficult to drop the disabled attribute on buttons wholesale and have a rule that says "all disabled buttons are focusable". The more ambiguous thing here is if we only allow them to be focusable in some situations but not in others.

**Comment by dzearing** [comment]
I've started a discussion internally with Edge and the Office Web Apps accessibility folks to make sure we're all following the same rules.

**Comment by adharpicstore** [comment]
@dzearing any update on this?
